### PR TITLE
Feat/pupil 833/pupil unit filter

### PR DESCRIPTION
**Pull request by benprotheroe**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

### Job Story

As a pupil squad engineer,
**I want to** see the filter units component in storybook
**So that [I / user] can** test it and use it easily in the pupil journey 

### Implementation details

https://www.figma.com/design/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?node-id=5536-197291&m=dev

Create an Oak component that represents the above with the following props

```jsx
menuItems={[{text:”All”,id:”0”}, {text:”Biology”,id:”1”}, {text:”Chemistry”,id:”2”}, {text:”Physics”,id:”3”}]}
selected={2}
onSelected={(menuItemId) ⇒ {}}
```

It must also overflow link this - 

https://www.figma.com/design/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?node-id=5646-7477&m=dev

On mobile there is an additional feature where the menu becomes expandable (via an Accordion, see below). It will be closed on mobile by default. The accordion should push/pull content down/up on the page.

### CODE REVIEW

File: src/components/atoms/InternalAccordion/InternalAccordion.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ */
+
+export const InternalAccordionContent = styled(UnstyledAccordionContent)``;
+
+const UnstyledAccordionButton = (
+  props: InternalAccordionProps & OakFlexProps & OakBoxProps,
+) => {
+  const { children, id, ...rest } = props;
+  const { isOpen, setOpen } = useAccordionContext();
```
Comment (kimon-satan) [inline src/components/atoms/InternalAccordion/InternalAccordion.tsx:73]: it's a nice idea to throw a helpful error for when someone forgot to use the provider . Like this example
```
  if (context === undefined) {
    throw new Error(`useCounter must be used within a CounterProvider`)
  }
  ```

File: src/components/atoms/InternalAccordion/useAccordionContext.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useContext } from "react";
```
Comment (kimon-satan) [inline src/components/atoms/InternalAccordion/useAccordionContext.tsx:1]: This file should be .ts  not .tsx .  If there is no HTML like tags then it's always .ts

File: src/components/atoms/InternalAccordion/AccordionProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { createContext, FC, useState } from "react";
```
Comment (kimon-satan) [inline src/components/atoms/InternalAccordion/AccordionProvider.tsx:1]: Should be called InternalAccordianProvider

File: src/components/molecules/OakHandDrawnFocusUnderline/HandDrawnFocusUnderlineSvg.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
```
Comment (kimon-satan) [inline src/components/molecules/OakHandDrawnFocusUnderline/HandDrawnFocusUnderlineSvg.tsx:1]: The svg probably should be in an assets folder. It should be called InternalHandDrawnFocusUnderline

Comment (kimon-satan) [reply src/components/molecules/OakHandDrawnFocusUnderline/HandDrawnFocusUnderlineSvg.tsx:1]: or if it's not even exported from the folder ... SubHandDrawnFocusUnderline

File: src/components/organisms/pupil/PupilJourneyUnitsFilter/PupilJourneyUnitsFilter.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React from "react";
```
Comment (kimon-satan) [inline src/components/organisms/pupil/PupilJourneyUnitsFilter/PupilJourneyUnitsFilter.stories.tsx:1]: Should be called OakPupilJourneyUnitsFilter

**Review by kimon-satan — CHANGES_REQUESTED** [review]

File: src/components/atoms/InternalAccordion/InternalAccordion.tsx (hunk 1 above)
Comment (benprotheroe) [reply src/components/atoms/InternalAccordion/InternalAccordion.tsx:73]: there is that see useAcordionContext component

**Review by kimon-satan — APPROVED** [review]

**Review by kimon-satan — APPROVED** [review]

**Comment by messmike** [comment]
[https://deploy-preview-228--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-pupiljourneyunitsfilter--docs](https://deploy-preview-228--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-pupiljourneyunitsfilter--docs)
@RBrownsett fyi

**Comment by messmike** [comment]
The chevron accordion icon looks a bit small, can the size be increased:
Screenshot 2024-07-15 at 11 11 49 am

**Comment by benprotheroe** [comment]
> The chevron accordion icon looks a bit small, can the size be increased: Screenshot 2024-07-15 at 11 11 49 am

good spot; thats done

**Review by RBrownsett — APPROVED** [review]
Approved - checked accessibility, mobile view and different browsers - no issues.

**Pull request merged**